src/redisearch_rs/inverted_index/src/lib.rs

```diff
@@ -54,6 +58,60 @@ impl IdDelta for u32 {
     }
 }
 
+/// Filter details to apply to numeric values
+/// cbindgen:rename-all=CamelCase
+#[repr(C)]
+pub struct NumericFilter {
+    field_spec: *const FieldSpec,
+
+    /// Beginning of the range
+    min: f64,
+
+    /// End of the range
+    max: f64,
+
+    /// Geo filter, if any
+    geo_filter: *const c_void,
+
+    /// Range includes the min value
+    min_inclusive: bool,
+
+    /// Range includes the max value
+    max_inclusive: bool,
+
+    /// Order of SORTBY (ascending/descending)
+    ascending: bool,
+
+    /// Minimum number of results needed
+    limit: usize,
+
+    /// Number of results to skip
+    offset: usize,
+}
+
+impl NumericFilter {
+    /// Check if this is a numeric filter (and not a geo filter)
+    pub fn is_numeric_filter(&self) -> bool {
+        self.geo_filter.is_null()
+    }
+
+    /// Check if the given value is in the range specified by this filter
+    pub fn value_in_range(&self, value: f64) -> bool {
+        let min_ok = if self.min_inclusive {
```

Comment:
any reason not to do these as `min_ok = value > self.min || (self.min_inclusive && value == self.min)`? Same for max. As I suppose it's more likely that you are greater or lesser, compared to the single possibility that you are equal

Reply:
No reason. I did not think of this approach but think it is worth making.